controllers/mealsController.js

```diff
@@ -49,6 +49,62 @@ const show = async (req, res) => {
   }
 }
 
+const index = async (req, res) => {
+  try {
+    const user = await findUser(req.body.api_key);
+    const meals = await findUserMeals(user.id);
+    const mealsResults = await findAllMealFoods(meals, user.id)
+    res.setHeader("Content-Type", "application/json");
+    res.status(201).send(JSON.stringify({ user_id: `${user.id}`, meals: mealsResults }) )
+  } catch (error) {
+    res.setHeader("Content-Type", "application/json");
+    res.status(400).send(JSON.stringify("Could not fetch meals."))
+  }
+}
+
+const findUserMeals = async (userId) => {
+  const mealFoods = await MealFood.findAll({ where: { UserId: userId }})
+  const meals = mealFoods.map(mealFood => {
+    const mealId = mealFood.MealId
+    const meal = Meal.findOne({ where: { id: mealId }})
+    return meal
+  })
+  const results = await Promise.all(meals)
+  const filteredResults = removeDuplicates(results)
+  return filteredResults
+}
+
+function removeDuplicates(meals) {
+  const result = [];
+  const map = new Map();
+  for (const item of meals) {
+    if(!map.has(item.id)){
+        map.set(item.id, true);
+        result.push({
+            id: item.id,
+            name: item.meal_name,
+            date: item.date
+        });
+    }
+  }
+  return result
+}
+
+const findAllMealFoods = (meals, userId) => {
+  const allMealFoods = meals.map(meal => {
+    return mealFoodsObject(meal, userId)
+  })
+  return Promise.all(allMealFoods)
```

Comment:
I'd love to get a second opinion, but as tedious as it was to figure out to use Promise.all here and above I'm pretty sure it means we're grabbing all the mealFoods at the same time. So that's good 😄